app/controllers/admin/lead_provider_users_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Admin::LeadProviderUsersController < ApplicationController
+  before_action :set_lead_provider
+
+  def index
+    @users = User.joins(:lead_provider_profile).where(lead_provider_profiles: { lead_provider: @lead_provider })
```

Comment:
That's where having `users through profiles` relation would be useful. You could do `@users = @lead_provider.users`.